src/hydrogen/device/rocBLAS.cpp

```diff
@@ -0,0 +1,141 @@
+#include <hydrogen/device/gpu/rocm/rocBLAS.hpp>
+
+// Helper macro for converting enums to strings.
+#define H_ADD_ROCBLAS_ENUM_TO_STRING_CASE(enum_value) \
+    case enum_value:                        \
+        return #enum_value
+
+
+namespace
+{
+
+std::string GetrocBLASErrorString(rocblas_status status)
+{
+    switch (status)
+    {
+        H_ADD_ROCBLAS_ENUM_TO_STRING_CASE(rocblas_status_success);
+        H_ADD_ROCBLAS_ENUM_TO_STRING_CASE(rocblas_status_invalid_handle);
+        H_ADD_ROCBLAS_ENUM_TO_STRING_CASE(rocblas_status_not_implemented);
+        H_ADD_ROCBLAS_ENUM_TO_STRING_CASE(rocblas_status_invalid_pointer);
+        H_ADD_ROCBLAS_ENUM_TO_STRING_CASE(rocblas_status_invalid_size);
+        H_ADD_ROCBLAS_ENUM_TO_STRING_CASE(rocblas_status_memory_error);
+        H_ADD_ROCBLAS_ENUM_TO_STRING_CASE(rocblas_status_internal_error);
+    default:
+        return "Unknown rocBLAS error.";
+    }
+}
+
+}
+
+namespace hydrogen
+{
+
+namespace rocblas
+{
+namespace // <anon2>
+{
+bool rocblas_is_initialized_ = false;
+rocblas_handle default_rocblas_handle_;
+}// namespace <anon2>
+
+rocblas_handle GetLibraryHandle() noexcept
+{
+    return default_rocblas_handle_;
+}
+
+bool IsInitialized() noexcept
+{
+    return rocblas_is_initialized_;
+}
+
+void Initialize(rocblas_handle handle)
+{
+    if (!IsInitialized())
+    {
+        if (!handle)
+            H_CHECK_ROCBLAS(rocblas_create_handle(&default_rocblas_handle_));
+        else
+            default_rocblas_handle_ = handle;
+
+        H_CHECK_ROCBLAS(
+            rocblas_set_stream(
+                GetLibraryHandle(), rocm::GetDefaultStream()));
+        H_CHECK_ROCBLAS(
+            rocblas_set_pointer_mode(
+                GetLibraryHandle(), rocblas_pointer_mode_host));
+
+        rocblas_is_initialized_ = true;
+    }
+}
+
+void Finalize()
+{
+    if (default_rocblas_handle_)
+        H_CHECK_ROCBLAS(rocblas_destroy_handle(default_rocblas_handle_));
```

Comment:
In case `Finalize` is called multiple times:
```suggestion
    if (default_rocblas_handle_)
        H_CHECK_ROCBLAS(rocblas_destroy_handle(default_rocblas_handle_));
    default_rocblas_handle_ = nullptr;
```